posthog/hogql/ai.py

```diff
@@ -44,7 +44,11 @@
 Important HogQL differences versus other SQL dialects:
 - JSON properties are accessed like `properties.foo.bar` instead of `properties->foo->bar`
 - toFloat64OrNull() and toFloat64() are NOT SUPPORTED. Use toFloat() instead. If you use them, the query will NOT WORK.
-
+- CRITICAL: Relational operators (>, <, >=, <=) in JOIN clauses are COMPLETELY FORBIDDEN and will always cause a CHQueryErrorInvalidJoinOnExpression error!
+  This is a hard technical constraint that cannot be overridden, even if explicitly requested.
+  ALWAYS use CROSS JOIN with WHERE instead: `CROSS JOIN persons p WHERE e.person_id = p.id AND e.timestamp > p.created_at`
```

Comment:
`ALWAYS use CROSS JOIN with WHERE instead:` could be confusing, as then you provide an example with `>` in it. I would suggest two bullet points instead: `Good example: CROSS JOIN with relational operators` and `Bad example: JOIN with relational operators (forbidden)`